### PR TITLE
Allowing pathlib.PosixPath params for yml files

### LINKED ISSUE #553: [Feature] Add support for Path when adding `warehouse_yml_file` and `scan_yml_file`
**Is your feature request related to a problem? Please describe.**
I was following the instructions for [programmatic scans](https://docs.soda.io/soda-sql/programmatic_scan.html) and when it came to entering a path to the `scan_yml_file` and `warehouse_yml_file` I entered a path using `from pathlib import Path`:
```
from pathlib import Path

ROOT_DIR = Path(__file__).parent.parent
scan_builder.warehouse_yml_file = ROOT_DIR / "soda_warehouse_directory" / "warehouse_local.yml"
```
 but I received an error: 
```
scan_builder.scan_yml_file must be str, but was <class 'pathlib.PosixPath'>: /home/my_user/my_repo/soda_warehouse_directory/tables/warehouse_local/my_scan.yml
```

It works when I just convert the Path to str, but supporting both would be nice since str is more common, but Path is nicer:
```
scan_builder.warehouse_yml_file = str(ROOT_DIR / "soda_warehouse_directory" / "warehouse_local.yml")
```

**Describe the solution you'd like**
Allow `Path` variables as well as `str` when entering a path to a file.

**Soda SQL Version**: 2.1.0b18

### PR DESCRIPTION
**Pull request by pgoslatara**
This PR resolves #553.

### CODE REVIEW

**Comment by vijaykiran** [comment]
@pgoslatara  Thank you for the contribution! can you please sign the CLA (and I've left a comment regarding posixpath)

File: core/sodasql/scan/scan_builder.py [hunk 1]
```diff
@@ -96,10 +97,15 @@ def _build_warehouse_yml(self):
             return
 
         elif self.warehouse_yml_file and not self.warehouse_yml_dict and not self.warehouse_yml:
-            if not isinstance(self.warehouse_yml_file, str):
-                logger.error(f'scan_builder.warehouse_yml_file must be str, but was {type(self.warehouse_yml_file)}: {self.warehouse_yml_file}')
+            if isinstance(self.warehouse_yml_file, pathlib.PosixPath):
```
[the remark this answers is not in the document]
Comment (pgoslatara) [reply core/sodasql/scan/scan_builder.py]: Good spot! Addressed in bd1b46f.

**Comment by pgoslatara** [comment]
@vijaykiran Can I get some help on understanding why these checks failed? It looks to me like the environment variables weren't set correctly in the "Setup Python 3.*" steps for some reason. I'm new to this so I don't understand why this happened or if there is something I can do to resolve this? Maybe because it's the second run and the `action_required` status affects setting env vars?

**Comment by vijaykiran** [comment]
@pgoslatara  I think there are some conflicts - I've cherry-picked your commit (added some additional type info)  and opened a new PR, once the build passes, I'll merge it.

https://github.com/sodadata/soda-sql/pull/573

**Pull request closed without merging**